homeassistant/components/demo/alarm_control_panel.py

```diff
@@ -31,7 +31,7 @@ async def async_setup_platform(
     """Set up the Demo alarm control panel platform."""
     async_add_entities(
         [
-            ManualAlarm(
+            ManualAlarm(  # type:ignore[no-untyped-call]
```

Comment:
I have had to add a few of these because `ManualAlarm`, `image_processing` and `mailbox` are untyped (at least partially)
I think it's still a good idea to add it to strict typing, to prevent new untyped functions making their way into the demo component.

Reply:
I'm not convinced adding this many type ignores just for strict typing is the right call. True adding types for `ManualAlarm.__init__` is slightly more difficult but all others are strait forward. There is no need to add types to the whole integrations, just the functions necessary to add strict typing here.

Besides that, there might be some (small?) value in **not** adding strict typing for `demo`. It will make interactions with untyped integrations slightly easier.

Reply:
I've marked as draft until #78022 is merged.

Reply:
> Besides that, there might be some (small?) value in **not** adding strict typing for `demo`. It will make interactions with untyped integrations slightly easier.

I disagree with that. IMO the benefit of having all checks enabled overrides the slight hassle of having to add `no-untyped-call` when adding a new platform.

Reply:
Ok, let's do it then.